api/common/v1alpha1/datasource/datasource.go

```diff
@@ -0,0 +1,63 @@
+// +kubebuilder:object:generate=true
+package datasource
+
+// +kubebuilder:validation:Enum=File;Secret;EnvVar;InsecureInline
+type SecureDataSourceType string
+
+const (
+	SecureDataSourceFile      SecureDataSourceType = "File"
+	SecureDataSourceSecretRef SecureDataSourceType = "Secret"
+	SecureDataSourceEnvVar    SecureDataSourceType = "EnvVar"
+	SecureDataSourceInline    SecureDataSourceType = "InsecureInline"
+)
+
+// SecureDataSource is a way to securely provide data to the component
+type SecureDataSource struct {
+	Type           SecureDataSourceType `json:"type,omitempty"`
+	File           *File                `json:"file,omitempty"`
+	EnvVar         *EnvVar              `json:"envVar,omitempty"`
+	InsecureInline *Inline              `json:"insecureInline,omitempty"`
+	SecretRef      *SecretRef           `json:"secretRef,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=File;EnvVar;Inline
+type DataSourceType string
+
+const (
+	DataSourceFile   DataSourceType = "File"
+	DataSourceEnvVar DataSourceType = "EnvVar"
+	DataSourceInline DataSourceType = "Inline"
+)
+
+// DataSource is just a way to provide data. Not necessarily secrets,
+// can be any data, i.e. certs, configs, OPA policies written in rego, lua plugins etc.
+type DataSource struct {
+	Type   DataSourceType `json:"type"`
+	File   *File          `json:"file,omitempty"`
+	EnvVar *EnvVar        `json:"envVar,omitempty"`
+	Inline *Inline        `json:"inline,omitempty"`
+}
+
+type File struct {
+	Path string `json:"path,omitempty"`
```

Comment:
When this field is used in a policy it will start getting linted by the api-linter and it won't pass. I think this field is required so it shouldn't have "omitempty" according to our [types doc](https://docs.google.com/document/d/1ZV8KaXr1kV4P5O-eq4gy5IxSMJQ4VzvTT-BmeBABuDo/edit?tab=t.0).

```suggestion
	Path string `json:"path"`
```